### PR TITLE
Email and privacy policy fields

### PR DESCRIPTION
**Pull request by rwd**
* Requires that a privacy policy be accepted before sending feedback
* Accepts the user's email address as a distinct field, and if provided includes it in the email body
* Adds a Feedback model to validate submissions
* Outputs per-attribute error messages in error responses

### CODE REVIEW

File: spec/models/feedback_spec.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  it { is_expected.to validate_presence_of(:page) }
+  it { is_expected.to validate_inclusion_of(:type).in_array(%w(comment correction bug)) }
+  it { is_expected.to validate_acceptance_of(:privacy_policy) }
+  it { is_expected.to validate_length_of(:text).is_at_most(400) }
+
+  it 'should validate that :text has at least 5 words' do
+    five_words = %w(one two three four five)
+    five_words.each_with_index do |_word, index|
+      words = five_words[0..index]
+      text = words.join(' ')
```
Comment (lbiedinger) [inline spec/models/feedback_spec.rb]: 'text' is only used in the assignment below, let's just do:
```
subject.text = words.join(' ')
```

File: lib/europeana/feedback_button.rb [hunk 2]
```diff
@@ -1,8 +1,35 @@
 # frozen_string_literal: true
+
 require 'europeana/feedback_button/engine'
 
 module Europeana
   module FeedbackButton
+    # TODO: why is this necessary?
```
Comment (lbiedinger) [inline lib/europeana/feedback_button.rb]: This doesn't seem to be necessary. Removing it 'worked' the same as with this present.

File: spec/dummy/config/application.rb [hunk 3]
```diff
@@ -3,19 +3,14 @@
...
-# require "active_storage/engine"
 require 'action_controller/railtie'
 require 'action_mailer/railtie'
 require 'action_view/railtie'
-# require "action_cable/engine"
-# require "sprockets/railtie"
-# require "rails/test_unit/railtie"
 
 Bundler.require(*Rails.groups)
+# TODO: why is this necessary?
```
Comment (lbiedinger) [inline spec/dummy/config/application.rb]: The Dummy App doesn't require 'europeana/feedback_button' in a gemfile or elsewhere.

**Review by lbiedinger — CHANGES_REQUESTED** [review]

File: app/views/concerns/europeana/feedback_button/feedbackable_view.rb [hunk 4]
```diff
@@ -13,6 +13,7 @@ def feedback
           form_action: europeana_feedback_button.feedback_path,
           maxlength: Europeana::FeedbackButton::Feedback::MAX_LENGTH,
           minwords: Europeana::FeedbackButton::Feedback::MIN_WORDS
+          privacy_policy_url: Europeana::FeedbackButton.privacy_policy_url
```
Comment (lbiedinger) [inline app/views/concerns/europeana/feedback_button/feedbackable_view.rb]: missing a comma on the line above.

**Review by lbiedinger — CHANGES_REQUESTED** [review]

File: app/views/europeana/feedback_button/feedback_mailer/post.text.erb [hunk 5]
```diff
@@ -1,4 +1,6 @@
 Type: <%= @type %>
 URL: <%= @page %>
+Email: <%= @email %>
+
 Text:
 <%= @text %>
```
Comment (lbiedinger) [inline app/views/europeana/feedback_button/feedback_mailer/post.text.erb:6]: let's do the concatenation here once the frontend starts sending the email and text separately.

Comment (rwd) [reply app/views/europeana/feedback_button/feedback_mailer/post.text.erb:6]: I'm not really sure we need concatenation do we? Doesn't the entire email body just end up in the ticket description in the JIRA service desk? (I haven't seen it, so not sure.)

Comment (lbiedinger) [reply app/views/europeana/feedback_button/feedback_mailer/post.text.erb:6]: sure, adding the email to the text part of the mail was based on the acceptance criteria, but I agree that it's sufficient if not better when formatted with the label as we do now.

**Review by lbiedinger — APPROVED** [review]

**Pull request merged**